addon/templates/components/nav-route.hbs

```diff
@@ -10,13 +10,20 @@
   }}
     <div class='content'>
       <div class='nav-route-icon'>
-        {{#if icon}}
-          {{frost-icon
-            hook=(concat hookPrefix '-link-icon')
-            hookQualifiers=hookQualifiers
-            pack=pack
-            icon=icon
-          }}
+        {{#if iconHref}}
+          <svg
+            class='frost-icon'
+            data-test={{hook (concat hookPrefix '-link-icon') hookQualifiers}}
+          >
+            <use xlink:href={{iconHref}} />
+          </svg>
```

Comment:
Did you investigate whether we would want to make `frost-icon` smarter to handle the URL instead of adding the logic here?

Reply:
I did not, as I didn't think we'd want to use external icons for anything other than an app-bar, at least, I didn't know of a use-case where we'd want that. Do you know of a reason we'd want to use an external `.svg` file for an icon outside of the app-bar?

Reply:
I am admittedly not as familiar with this corner of things as I believe you and others are, but even if this is the only case in which there may be such a need shouldn't icon-related pathing code be placed with the icon code proper?   Then if another use case does arise it is ready for shared consumption.

Reply:
I suppose. But I didn't think we'd want to encourage that, as using an external icon rather than an icon pack would go against the design of `frost-icon` as far as I understood it. I would be worried that adding that functionality to `frost-icon` would encourage people to use that path instead of the more efficient and supported way of providing icons, using icon packs. 

If you think `frost-icon` should support external icons, I'm fine with the change being there, I just didn't want to encourage your users to use your components in ways other than intended, and I didn't think you'd want `frost-icon` to support external svgs. Perhaps I was mistaken.

Reply:
Yeah, if you want the change in `frost-icon`, that's ez-pz. Like Adam, I figured it would better to address this edge case separately so as to not encourage people to break away from icon packs.

Reply:
I understand what you are getting at (here and at https://github.com/ciena-frost/ember-frost-navigation/pull/221#issuecomment-344059755) and it makes me think we should be taking a larger view of how to serve dynamic icons (perhaps via dynamic icon packs) but that is certainly something we can circle back around to at a later time.  I will open an issue to revisit this portion of the conversation.